Document SO migrations enableV2 and batchSize config options (#96290)
elastic/kibana: docs/setup/settings.asciidoc

override this parameter to use their own Tile Map Service. For example:
`"https://tiles.elastic.co/v2/default/{z}/{x}/{y}.png?elastic_tile_service_tos=agree&my_app_name=kibana"`

| `migrations.batchSize:`

>  | Defines the number of documents migrated at a time. The higher the value, the faster the Saved Objects migration process will be performed at the cost of higher memory consumption. Should migration fail with the `circuit_breaking_exception` exception, you need to set a smaller `batchSize` value. *Default: `1000`*

Reviewer: nit: 

```suggestion
 | Defines the number of documents migrated at a time. The higher the value, the faster the Saved Objects migration process will be performed at the cost of higher memory consumption. Should the migration fail with a `circuit_breaking_exception`, you need to set a smaller `batchSize` value. *Default: `1000`*
```

Could also consider linking out to the circuit breaker docs https://www.elastic.co/guide/en/elasticsearch/reference/current/circuit-breaker.html

| `migrations.enableV2:`

>  | experimental[]. Enables the new Saved Objects migration algorithm. See <<upgrade-migrations>> for details. The setting is going to removed from `v8.0` when `migrations v2` becomes stable. *Default: `true`*

Reviewer: nit: Could consider explicitly saying something along the lines of "disabling this setting will cause Kibana to use the legacy migration algorithm which shipped in 7.11 and earlier", since reverting to legacy is the only reason someone would need to touch this.

>  | experimental[]. Enables the new Saved Objects migration algorithm. See <<upgrade-migrations>> for details. The setting is going to removed from `v8.0` when `migrations v2` becomes stable. Setting the value to `false` will cause Kibana to use the legacy migration algorithm, which shipped in 7.11 and earlier versions. *Default: `true`*

Reviewer 2: V2 migrations isn't a breaking change and keeping both adds technical debt so I think we should remove it as soon as possible.

Right now we've only seen v2 migrations fail because of other bugs which caused unusual amounts of saved objects and even though we have the `enableV2: false` workaround we've been asking users to clean up the unnecessary objects instead.

So after 7.12.1 I don't think we would need to use v1 migrations as a backup anymore, but just incase we might decide to keep it in 7.13 and remove in 7.14 (worst case we can always postpone this to later).

If you agree, maybe we should also deprecate this configuration option? I can create an issue to track this for the future.

override this parameter to use their own Tile Map Service. For example:
`"https://tiles.elastic.co/v2/default/{z}/{x}/{y}.png?elastic_tile_service_tos=agree&my_app_name=kibana"`

| `migrations.batchSize:`

>  | Defines the number of documents migrated at a time. The higher the value, the faster the Saved Objects migration process will be performed at the cost of higher memory consumption. Should the migration fail with a `circuit_breaking_exception`, you need to set a smaller `batchSize` value. *Default: `1000`*

Reviewer 3: ```suggestion
 | Defines the number of documents migrated at a time. The higher the value, the faster the Saved Objects migration process performs at the cost of higher memory consumption. If the migration fails due to a `circuit_breaking_exception`, set a smaller `batchSize` value. *Default: `1000`*
```

| `migrations.enableV2:`

>  | experimental[]. Enables the new Saved Objects migration algorithm. See <<upgrade-migrations>> for details. The setting is going to be removed in an upcoming release without any further notice when `migrations v2` becomes stable. Setting the value to `false` will cause Kibana to use the legacy migration algorithm, which shipped in 7.11 and earlier versions. *Default: `true`*

Reviewer 3: ```suggestion
 | experimental[]. Enables the new Saved Objects migration algorithm. For information about the migration algorithm, refer to <<upgrade-migrations>>. When `migrations v2` is stable, the setting will be removed in an upcoming release without any further notice. Setting the value to `false` causes {kib} to use the legacy migration algorithm, which shipped in 7.11 and earlier versions. *Default: `true`*
```

| `migrations.retryAttempts:`

>  | The number of times migrations will retry temporary failures such as a network timeout, 503 status code or `snapshot_in_progress_exception`. If upgrade migrations frequently fail after exhausting all retry attempts with a message like "Unable to complete the [...] step after 15 attempts, terminating." you can try to increase this setting. *Default: `15`*

Reviewer 3: ```suggestion
 | The number of times migrations retry temporary failures, such as a network timeout, 503 status code, or `snapshot_in_progress_exception`. When upgrade migrations frequently fail after exhausting all retry attempts with a message such as `Unable to complete the [...] step after 15 attempts, terminating.`, increase the setting value. *Default: `15`*
```
